src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -49,6 +49,24 @@ public enum JsonTokenType : byte
         String = (byte)6,
         True = (byte)8,
     }
+    public sealed partial class JsonWriterException : System.Exception
```

Comment:
Have we considered putting reader and writer in two separate subnamespaces? I cannot think of many (any?) scenarios where a developer would use both at the same time (in the same file).

Reply:
> any?) scenarios

Processing a JSON web request input and sending a JSON web response output?

Transforming JSON data to sanitize it in some way (e.g. strip comments)?

Searching a JSON file for a particular node and creating new JSON output containing only that subset?

Reply:
A few Json libs (e.g. utf8json, spanjson) use a transform approach for pretty-printing (assuming it's not really performance critical) and while-loop reading tokens, writing the tokens out again and add a few newlines/whitespaces here and there.